crates/sui-json-rpc-types/src/sui_transaction.rs

```diff
@@ -638,56 +638,43 @@ impl TryFrom<TransactionEffects> for SuiTransactionBlockEffects {
     type Error = SuiError;
 
     fn try_from(effect: TransactionEffects) -> Result<Self, Self::Error> {
-        let message_version = effect
-            .message_version()
-            .expect("TransactionEffects defines message_version()");
-
-        match message_version {
-            1 => Ok(SuiTransactionBlockEffects::V1(
-                SuiTransactionBlockEffectsV1 {
-                    status: effect.status().clone().into(),
-                    executed_epoch: effect.executed_epoch(),
-                    modified_at_versions: effect
-                        .modified_at_versions()
+        Ok(SuiTransactionBlockEffects::V1(
```

Comment:
Because we don't have SuiTransactionBlockEffectsV2 yet. It's not clear to me what are the steps involved to add SuiTransactionBlockEffectsV2 to the RPC layer. But I think we could ship this to core first and then look at RPC layer. I am also curious how this overlaps with the graphql timeline.

Reply:
I'd hope SDK would hide these stuff